kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -113,7 +124,9 @@ public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentW
 
         // if we get here, we have successfully rebased (i.e no logical conflicts)
         // against the winning transactions
-        return new TransactionRebaseState(getLastWinningTxnVersion(winningCommits));
+        return new TransactionRebaseState(
+                getLastWinningTxnVersion(winningCommits),
+                getLastCommitTimestamp(engine, winningCommits, winningCommitInfoOpt.get()));
```

Comment:
I see that `getLastWinningTxnVersion` is being invoked inside `getLastCommitTimestamp` as well. Can you store it in variable and pass it to both places from here instead?